src/util/config.cpp

```diff
@@ -114,75 +123,92 @@ bool configt::set(const cmdlinet &cmdline, const messaget &msg)
   if(cmdline.isset("fixedbv"))
     ansi_c.use_fixed_for_float = true;
 
+  // this is the default
+  const char *arch = "x86_64", *os = this_operating_system();
```

Comment:
We can use std::string here.

Reply:
We can, I just wanted to avoid unnecessary dynamic allocation when one of the options changing these values is set. It will be converted to std::string in `triple`. It's not critical at all, I'll change it...

Reply:
Well, it's a small string so it shouldn't be too problematic... My main concern is to keep different interfaces in the API for similar features, i.e., `this_architecture` returns a string while `this_operating_system` now returns a const char*.

Reply:
I also thought about changing `this_architecture()`, but there was no need to touch it (as we still default to `x86_64` regardless of the host -- mainly since the help message says so). To me it's not entirely clear whether that is the intended way: should for example the default on aarch64 be aarch64 as well?

Reply:
I think the main reason was that we focused mainly on x86_64 in the past; only recently we started to investigate other archs despite having some rudimentary support for them.

I would expect that the default behaviour was to default to the host arch.

Regarding the string return type, we can also make them `constexpr const char*`.

Reply:
> Regarding the string return type, we can also make them `constexpr const char*`.

Not without exposing the implementation in the header (constexpr implies inline). I'll leave them as std::string-s.